packages/nvidia_nat_core/src/nat/runtime/user_manager.py

```
@@ -0,0 +1,243 @@
+        """Extract the ``nat-session`` cookie value from a Request or WebSocket."""
+        if isinstance(connection, Request):
+            cookies: dict[str, str] = dict(connection.cookies) if connection.cookies else {}
+            return cookies.get(SESSION_COOKIE_NAME)
+
+        if isinstance(connection, WebSocket) and hasattr(connection, "scope") and "headers" in connection.scope:
+            for name, value in connection.scope.get("headers", []):
+                try:
+                    name_str: str = name.decode("utf-8").lower()
+                    value_str: str = value.decode("utf-8")
+                except Exception:
+                    continue
```

Ruff (0.15.5)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)